README.md

```diff
@@ -181,6 +181,112 @@ If you want to compile with ROCm support, install
+### Prerequisites
+- Ubuntu OS
+- NVIDIA GPU
+- Basic knowledge of using the terminal
+### Step 1: Update and Upgrade Ubuntu
+Update and upgrade your Ubuntu system to ensure you have the latest packages and security updates.
+```bash
+sudo apt update
+sudo apt dist-upgrade
+```
+### Step 2: Set Up Downloads Directory (Optional)
+Create a symbolic link to your Downloads directory if you have one on your Windows system.
+```bash
+ln -s /mnt/c/Users/<user>/Downloads Downloads
+```
+Replace `<user>` with your Windows username.
+### Step 3: Install Anaconda
+Download Anaconda and install it. Make sure to give execute permissions to the downloaded Anaconda script.
+```bash
+wget https://repo.anaconda.com/archive/Anaconda3-2024.02-1-Linux-x86_64.sh
+chmod +x /home/<user>/Downloads/Anaconda3-2024.02-1-Linux-x86_64.sh
+./home/<user>/Downloads/Anaconda3-2024.02-1-Linux-x86_64.sh
+```
+Activate the newly installed Anaconda environment and create a new environment for PyTorch development.
+```bash
+exec bash
+conda create --name pytorch-dev python=3.12
+conda activate pytorch-dev
+```
+Install necessary Python packages for PyTorch.
+```bash
+conda install astunparse numpy ninja pyyaml mkl mkl-include setuptools cmake cffi typing_extensions future six requests dataclasses
+```
+### Step 4: Install NVIDIA CUDA Toolkit
+Install CUDA on your Windows machine from the 12.4.1 repo. Then, download and install the NVIDIA CUDA toolkit on Ubuntu.
+```bash
+wget https://developer.download.nvidia.com/compute/cuda/repos/wsl-ubuntu/x86_64/cuda-keyring_1.1-1_all.deb
+sudo dpkg -i cuda-keyring_1.1-1_all.deb
+sudo apt-get update
+sudo apt-get -y install cuda
+```
+### Step 5: Set CUDA Paths
+Set the CUDA paths to enable CUDA in your system. I've added these lines to my `.bashrc` file, so they persist after a reboot see below instuctions.
+```bash
+export PATH=/usr/local/cuda/bin/:$PATH
+export LD_LIBRARY_PATH=/usr/local/cuda/lib64:$LD_LIBRARY_PATH
+```
+Verify the CUDA installation. If this line does not give an output, it means the paths are not set correctly from the export commands above. You may not have linked the binaries with the export commands or updated the source to reflect the changes in the `.bashrc` file. A simple reboot may solve it.
+```bash
+nvcc --version
+```
+### Step 6: Install cuDNN
+Download and install cuDNN from the NVIDIA repository.
+```bash
+wget https://developer.download.nvidia.com/compute/cudnn/9.1.0/local_installers/cudnn-local-repo-ubuntu2004-9.1.0_1.0-1_amd64.deb
+sudo dpkg -i cudnn-local-repo-ubuntu2004-9.1.0_1.0-1_amd64.deb
+sudo cp /var/cudnn-local-repo-ubuntu2004-9.1.0/cudnn-*-keyring.gpg /usr/share/keyrings/
+sudo apt-get update
+sudo apt-get -y install cudnn-cuda-12
```

Comment:
This seems a bit strange that you want to install CUDNN as Debian package, but rest of the toolkit is installed via local installer.

Reply:
I first tried the local installer and it failed for whatever reason. so I tried the Deb package  and it worked. the tutorial  was based on my bash history from working commands ommiting any failures. I'm sure it will work on others pc the other way. yet in my case it didn't  work. so I moved on.

Reply:
i will amend to sudo apt-get -y install cudnn-cuda-12 so that it just installed form the previous Nvidia apt key that was added in prior step.

Reply:
in a new environment and fresh ubuntu install i tried to do it the way you suggested with a network install from the same repo as we loaded above. and it is not in the wsl repo cudnn so i have had to do it again via the deb installer. i added this to explain wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/cuda-keyring_1.1-1_all.deb
sudo dpkg -i cuda-keyring_1.1-1_all.deb
sudo add-apt-repository "deb https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/ /"
sudo apt-get update
sudo apt-get -y install cuda-12-4
sudo apt-get install bzip2-doc cpp-doc gcc-11-locales g++-multilib bison gdb gcc-doc libasound2-plugins alsa-utils mesa-utils
#or
sudo apt-get install cuda-12.4 --install-suggests
# as per the cuDNN Downloads instructions i next tried to do a network install using this :
#wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/cuda-keyring_1.1-1_all.deb
#sudo dpkg -i cuda-keyring_1.1-1_all.deb
#sudo apt-get update
#sudo apt-get -y install cudnn
#however as you can see this will override the cuda-keyring_1.1-1_all.deb that was installed via wsl repo above.
#if you try running this command directly sudo apt-get -y install cudnn it does not work with the wsl repo of cuda.
#so i instead ran these commands to sintall it directly  via a local file
wget https://developer.download.nvidia.com/compute/cudnn/9.1.0/local_installers/cudnn-local-repo-ubuntu2004-9.1.0_1.0-1_amd64.deb
sudo dpkg -i cudnn-local-repo-ubuntu2004-9.1.0_1.0-1_amd64.deb
sudo cp /var/cudnn-local-repo-ubuntu2004-9.1.0/cudnn-*-keyring.gpg /usr/share/keyrings/
sudo apt-get update
sudo apt-get -y install cudnn